### PR TITLE
docs: Add windows publishing steps

### PR DESCRIPTION
**Pull request by jeromelaban**
related to https://github.com/unoplatform/uno/issues/14385 and https://github.com/unoplatform/private/issues/400, closes https://github.com/unoplatform/uno/issues/16662

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: doc/articles/toc.yml [hunk 1]
```diff
@@ -231,6 +231,13 @@
         topicHref: guides/silverlight-migration/silverlight-migration-landing.md
         href: guides/silverlight-migration/toc.yml
 
+    - name: Publishing
+      items:
+        - name: Overview
+          href: xref:uno.publishing.overview
+        - name: Publishing for Windows
+          href: xref:uno.publishing.windows
```
Comment (agneszitte) [inline doc/articles/toc.yml:239]:
```suggestion
          href: xref:uno.publishing.windows
```

File: doc/articles/uno-publishing-windows.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+uid: uno.publishing.windows
+---
+
+# Publishing Your App for Windows
+
+## Preparing For Publish
+
+## Building your app
+
+Building your app require using the msbuild command (`dotnet build` is not compatible as of WinAppSDK 1.5).
```
Comment (agneszitte) [inline doc/articles/uno-publishing-windows.md:11]:
```suggestion
Building your app requires using the msbuild command (`dotnet build` is not compatible as of WinAppSDK 1.5).
```

File: doc/articles/uno-publishing-windows.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  <Identity Name="MyApp"
+            Publisher="CN=AppModelSamples, OID.2.25.311729368913984317654407730594956997722=1"
+            Version="1.0.0.0" />
+  ```
+
+- Build your app in the folder of the app's `.csproj` by using the following command:
+
+    ```pwsh
+    msbuild /r /p:TargetFramework=net8.0-windows10.0.19041 /p:Configuration=Release /p:Platform=x64 /p:GenerateAppxPackageOnBuild=true /p:AppxBundle=Never /p:UapAppxPackageBuildMode=Sideloading /p:AppxPackageDir="C:/temp/output/" /p:AppxPackageSigningEnabled=false
+    ```
```
Comment (agneszitte) [inline doc/articles/uno-publishing-windows.md:33]:
```suggestion
 ```pwsh
msbuild /r /p:TargetFramework=net8.0-windows10.0.19041 /p:Configuration=Release /p:Platform=x64 /p:GenerateAppxPackageOnBuild=true /p:AppxBundle=Never /p:UapAppxPackageBuildMode=Sideloading /p:AppxPackageDir="C:/temp/output/" /p:AppxPackageSigningEnabled=false
    ```
```

File: doc/articles/uno-publishing-windows.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+> [!IMPORTANT]
+> Single package msix bundling is [not yet supported from msbuild the command line](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/single-project-msix?tabs=csharp#automate-building-and-packaging-your-single-project-msix-app).
+
+#### Install unsigned the app
+
+To install the app:
+
+- Start an elevated powershell command prompt (Search for powershell in the start menu, right click on it then Run as administrator):
+- In the folder contaning the `.msix` file, execute the following command :
```
Comment (agneszitte) [inline doc/articles/uno-publishing-windows.md:43]:
```suggestion
To install the app:

- Start an elevated PowerShell command prompt (Search for PowerShell in the start menu, right-click on it then Run as administrator):
- In the folder containing the `.msix` file, execute the following command :
```

File: doc/articles/uno-publishing-windows.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  - Open the solution in Visual Studio 2022
+  - Double click on the `Package.appxmanifest` file
+  - Navigate to the `Packaging` tab
+  - Click the **Create** button, then set a **publisher common name**, then OK. Do not se  password.
+  - Close the **Choose a Certificate** window by clicking OK.
+  - Click the **Save file** button in the Visual Studio toolbar
+- Build the app on the command line in the folder of the app's `.csproj`, use the following command:
+
+    ```cmd
+    msbuild /r /p:TargetFramework=net8.0-windows10.0.19041 /p:Configuration=Release /p:Platform=x64 /p:GenerateAppxPackageOnBuild=true /p:AppxBundlePlatforms="x86|x64|arm64" /p:AppxBundle=Never /p:UapAppxPackageBuildMode=Sideloading /p:AppxPackageDir="C:\temp\output\\" /p:AppxPackageSigningEnabled=true
```
Comment (agneszitte) [inline doc/articles/uno-publishing-windows.md:69]:
```suggestion
### Side loading (with signing)

#### Package signed the app

This guide uses a self-signed certificate.

To package your app:

- Create a self-signed certificate:
  - Open the solution in Visual Studio 2022
  - Double-click on the `Package.appxmanifest` file
  - Navigate to the `Packaging` tab
  - Click the **Create** button, then set a **publisher common name**, then OK. Do not set a password.
  - Close the **Choose a Certificate** window by clicking OK.
  - Click the **Save file** button in the Visual Studio toolbar
- Build the app on the command line in the folder of the app's `.csproj`, use the following command:

```cmd
msbuild /r /p:TargetFramework=net8.0-windows10.0.19041 /p:Configuration=Release /p:Platform=x64 /p:GenerateAppxPackageOnBuild=true /p:AppxBundlePlatforms="x86|x64|arm64" /p:AppxBundle=Never /p:UapAppxPackageBuildMode=Sideloading /p:AppxPackageDir="C:\temp\output\\" /p:AppxPackageSigningEnabled=true
```

Comment (agneszitte) [reply doc/articles/uno-publishing-windows.md:69]: Also @jeromelaban, where is the "Create" button you are mentioning?
image

File: doc/articles/uno-publishing-windows.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    ```
+
+In order to build for additional platforms, change the `Platform` parameter to `x86` or `arm64` to create additional MSIX
+
+#### Install signed the app
+
+To install the app:
+
+- Install the certificate on the machine:
+  - In the file explorer, right click on the `.msix` file, then **Show More Options** on Windows 11, then **Properties**
```
Comment (agneszitte) [inline doc/articles/uno-publishing-windows.md:79]:
```suggestion
  - In the file explorer, right-click on the `.msix` file, then **Show More Options** on Windows 11, then **Properties**
```

File: doc/articles/uno-publishing-windows.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+  - Open the **Digital signatures** tab
+  - Click on your certificate then **Details**
+  - Click the **View Certificate** button
+  - Click the **Install Certificate** button
+  - Select **Local Machine** then **Next**
+  - Click **Place all certificates in the following store** then **Browse**
+  - Select **Trusted People** then **OK**
+  - Click **Next**, then **Finish**
+  - Close all the opened properties windows.
+- Install the `.msix` by double clicking on it
```
Comment (agneszitte) [inline doc/articles/uno-publishing-windows.md:89]:
```suggestion
- Install the `.msix` by double-clicking on it
```

File: doc/articles/uno-publishing-windows.md (hunk 5 above)
Comment (jeromelaban) [reply doc/articles/uno-publishing-windows.md:69]: Ah it's missing the Choose Certificate button step :) Will add.

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.2

File: doc/articles/uno-publishing-windows.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+#### Package signed the app
+
+This guide uses a self-signed certificate.
+
+To package your app:
+
+- Create a self-signed certificate:
+  - Open the solution in Visual Studio 2022
+  - Double-click on the `Package.appxmanifest` file
```
Comment (agneszitte) [inline doc/articles/uno-publishing-windows.md:61]: @jeromelaban not sure if you can replicate this issue on your side
But with an Uno Platform application when double-clicking on the `Package.appxmanifest` file, most of the time I have this error. 
image
I tried with a WinUI application multiple times and I don't have this error.
______
Using Microsoft Visual Studio Professional 2022 (64-bit) - Current Version 17.9.6 for my tests

Comment (MartinZikmund) [reply doc/articles/uno-publishing-windows.md:61]: Yes, I remember this happened to me as well in single project

Comment (jeromelaban) [reply doc/articles/uno-publishing-windows.md:61]: I'm not having that problem, tried on multiple machines, stable and preview. @MartinZikmund could you give it a try please?

File: doc/articles/uno-publishing-windows.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+  ```xml
+  <Identity Name="MyApp"
+            Publisher="CN=AppModelSamples, OID.2.25.311729368913984317654407730594956997722=1"
+            Version="1.0.0.0" />
+  ```
+
+- Build your app in the folder of the app's `.csproj` by using the following command:
+
+    ```pwsh
+    msbuild /r /p:TargetFramework=net8.0-windows10.0.19041 /p:Configuration=Release /p:Platform=x64 /p:GenerateAppxPackageOnBuild=true /p:AppxBundle=Never /p:UapAppxPackageBuildMode=Sideloading /p:AppxPackageDir="C:/temp/output/" /p:AppxPackageSigningEnabled=false
```
Comment (MartinZikmund) [inline doc/articles/uno-publishing-windows.md:32]: This should also include docs on how to make a bundle of x64, ARM64 and x86 packages (for Store publishing)

File: doc/articles/uno-publishing-windows.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+  - Click the **Create** button, then set a **publisher common name**, then OK. Do not set a password.
+  - Close the **Choose a Certificate** window by clicking OK.
+  - Click the **Save file** button in the Visual Studio toolbar
+- Build the app on the command line in the folder of the app's `.csproj`, use the following command:
+
+  ```cmd
+  msbuild /r /p:TargetFramework=net8.0-windows10.0.19041 /p:Configuration=Release /p:Platform=x64 /p:GenerateAppxPackageOnBuild=true /p:AppxBundlePlatforms="x86|x64|arm64" /p:AppxBundle=Never /p:UapAppxPackageBuildMode=Sideloading /p:AppxPackageDir="C:/temp/output/" /p:AppxPackageSigningEnabled=true
+  ```
+
+In order to build for additional platforms, change the `Platform` parameter to `x86` or `arm64` to create additional MSIX
```
Comment (MartinZikmund) [inline doc/articles/uno-publishing-windows.md:73]: The command contains `AppxBundlePlatforms` for all the platforms - but the build will only produce x64, correct? Will omitting /p:Platform produce the full bundle?

Comment (jeromelaban) [reply doc/articles/uno-publishing-windows.md:73]: This parameter is like a red herring, caused by another issue where resizetizer causes some targets to be executed incorrectly.  (https://github.com/unoplatform/uno/discussions/16645)

File: doc/articles/uno-publishing-windows.md (hunk 8 above)
Comment (Darsh0307) [reply doc/articles/uno-publishing-windows.md:71]: Hey @jeromelaban  , i did had this issue when i was trying to open the file
Templates version :- 5.2.31
VSIX version :- 5.2.48
VS version :- 17.9.6

Comment (Darsh0307) [reply doc/articles/uno-publishing-windows.md:71]: Screenshot 2024-05-14 at 11 56 32 AM

Comment (jeromelaban) [reply doc/articles/uno-publishing-windows.md:71]: This is the issue: https://developercommunity.visualstudio.com/t/Double-clicking-on-a-PackageAppxmanifes/10658683. Opening the file requires to be in ios, android, catalyst or windows target frameworks. The other target frameworks produce the error.

File: doc/articles/uno-publishing-windows.md [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+- Build your app in the folder of the app's `.csproj` by using the following command:
+
+    ```pwsh
+    msbuild /r /p:TargetFramework=net8.0-windows10.0.19041 /p:Configuration=Release /p:Platform=x64 /p:GenerateAppxPackageOnBuild=true /p:AppxBundle=Never /p:UapAppxPackageBuildMode=Sideloading /p:AppxPackageDir="C:/temp/output/" /p:AppxPackageSigningEnabled=false
+    ```
+
+In order to build for additional platforms, change the `Platform` parameter to `x86` or `arm64` to create additional MSIX files.
+
+> [!IMPORTANT]
+> IF your solution contains `net8.0` projects, you will need to split the above build command in two, one to restore NuGet packages, the other one to create the package.
```
Comment (agneszitte) [inline doc/articles/uno-publishing-windows.md:38]:
```suggestion
> IF your solution contains `net8.0` projects, you will need to split the above build command in two, one to restore NuGet packages, and the other one to create the package.
```

File: doc/articles/uno-publishing-windows.md [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+>
+> First run this command:
+>
+> ```pwsh
+> msbuild /r /t:Restore /p:Configuration=Release
+> ```
+>
+> Then run the package creation command without the `/r` parameter.
+
+#### Install unsigned the app
```
Comment (agneszitte) [inline doc/articles/uno-publishing-windows.md:48]:
```suggestion
#### Install the unsigned app
```

File: doc/articles/uno-publishing-windows.md [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+  - Click the **Create** button, then set a **publisher common name**, then OK. Do not set a password.
+  - Close the **Choose a Certificate** window by clicking OK.
+  - Click the **Save file** button in the Visual Studio toolbar
+- Build the app on the command line in the folder of the app's `.csproj`, use the following command:
+
+  ```cmd
+  msbuild /r /p:TargetFramework=net8.0-windows10.0.19041 /p:Configuration=Release /p:Platform=x64 /p:GenerateAppxPackageOnBuild=true /p:AppxBundle=Never /p:UapAppxPackageBuildMode=Sideloading /p:AppxPackageDir="C:/temp/output/" /p:AppxPackageSigningEnabled=true
+
+> [!IMPORTANT]
+> IF your solution contains `net8.0` projects, you will need to split the above build command in two, one to restore NuGet packages, the other one to create the package.
```
Comment (agneszitte) [inline doc/articles/uno-publishing-windows.md:83]:
```suggestion
> IF your solution contains `net8.0` projects, you will need to split the above build command in two, one to restore NuGet packages, and the other one to create the package.
```

File: doc/articles/uno-publishing-windows.md [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+> ```
+>
+> Then run the package creation command without the `/r` parameter.
+
+In order to build for additional platforms, change the `Platform` parameter to `x86` or `arm64` to create additional MSIX files.
+
+> [!IMPORTANT]
+> Single package msix bundling is [not yet supported from msbuild the command line](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/single-project-msix?tabs=csharp#automate-building-and-packaging-your-single-project-msix-app).
+
+#### Install signed the app
```
Comment (agneszitte) [inline doc/articles/uno-publishing-windows.md:98]:
```suggestion
#### Install the signed app
```

File: doc/articles/uno-publishing-windows.md (hunk 10 above)
Comment (jeromelaban) [reply doc/articles/uno-publishing-windows.md]: Removed in latest version of the doc.

File: doc/articles/uno-publishing-windows.md (hunk 8 above)
Comment (jeromelaban) [reply doc/articles/uno-publishing-windows.md:72]: VS Issue: https://developercommunity.visualstudio.com/t/Double-clicking-on-a-PackageAppxmanifes/10658683

File: doc/articles/uno-publishing-windows-packaged-signed.md [hunk 15]
```diff
@@ -0,0 +1,87 @@
+---
+uid: uno.publishing.windows.sideload.packaged.signed
+---
+
+# Build a signed packaged app
+
+This guide will show how to create an signed packaged app using Windows App SDK.
```
Comment (agneszitte) [inline doc/articles/uno-publishing-windows-packaged-signed.md:7]:
```suggestion
This guide will show how to create a signed packaged app using Windows App SDK.
```

File: doc/articles/uno-publishing-windows-packaged-signed.md [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+  ```
+
+In order to build for additional platforms, change the `Platform` parameter to `x86` or `arm64` to create additional MSIX files.
+
+> [!IMPORTANT]
+> Single package msix bundling is [not yet supported from msbuild the command line](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/single-project-msix?tabs=csharp#automate-building-and-packaging-your-single-project-msix-app).
+
+## Considerations for solutions with class library projects
+
+If your app references multiple library projects projects, you will need to split the above build command in two parts, one to restore NuGet packages, and the other one to create the package.
```
Comment (agneszitte) [inline doc/articles/uno-publishing-windows-packaged-signed.md:41]:
```suggestion
If your app references multiple library projects, you will need to split the above build command into two parts, one to restore NuGet packages, and the other one to create the package.
```

File: doc/articles/uno-publishing-windows-packaged-signed.md [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+  ```xml
+  <PropertyGroup Condition=" '$(PublishSignedPackage)' == 'true' ">
+    <GenerateAppxPackageOnBuild>true</GenerateAppxPackageOnBuild>
+    <AppxBundle>Never</AppxBundle>
+    <UapAppxPackageBuildMode>Sideloading</UapAppxPackageBuildMode>
+    <AppxPackageSigningEnabled>true</AppxPackageSigningEnabled>  
+  </PropertyGroup>
+  ```
+
+- Run this command to restore the nuget packages:
```
Comment (agneszitte) [inline doc/articles/uno-publishing-windows-packaged-signed.md:56]:
```suggestion
- Run this command to restore the Nuget packages:
```

File: doc/articles/uno-publishing-windows-packaged-unsigned.md [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+
+    ```pwsh
+    msbuild /r /p:TargetFramework=net8.0-windows10.0.19041 /p:Configuration=Release /p:Platform=x64 /p:GenerateAppxPackageOnBuild=true /p:AppxBundle=Never /p:UapAppxPackageBuildMode=Sideloading /p:AppxPackageDir="C:/temp/output/" /p:AppxPackageSigningEnabled=false
+    ```
+
+In order to build for additional platforms, change the `Platform` parameter to `x86` or `arm64` to create additional MSIX files.
+
+## Considerations for solutions with class library projects
+
+If your app references multiple library projects projects, you will need to split the above build command in two parts, one to restore NuGet packages, and the other one to create the package.
```
Comment (agneszitte) [inline doc/articles/uno-publishing-windows-packaged-unsigned.md:37]:
```suggestion
If your app references multiple library projects, you will need to split the above build command into two parts, one to restore NuGet packages, and the other one to create the package.
```

File: doc/articles/uno-publishing-windows-packaged-unsigned.md [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+  ```xml
+  <PropertyGroup Condition=" '$(PublishUnsignedPackage)' == 'true' ">
+    <GenerateAppxPackageOnBuild>true</GenerateAppxPackageOnBuild>
+    <AppxBundle>Never</AppxBundle>
+    <UapAppxPackageBuildMode>Sideloading</UapAppxPackageBuildMode>
+    <AppxPackageSigningEnabled>false</AppxPackageSigningEnabled>  
+  </PropertyGroup>
+  ```
+
+- Run this command to restore the nuget packages:
```
Comment (agneszitte) [inline doc/articles/uno-publishing-windows-packaged-unsigned.md:52]:
```suggestion
- Run this command to restore the Nuget packages:
```

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.2

**Pull request merged**